src/Sylius/Component/Grid/Provider/ArrayGridProvider.php

```diff
@@ -30,20 +40,35 @@
      */
     public function __construct(ArrayToDefinitionConverterInterface $converter, array $gridConfigurations)
     {
-        foreach ($gridConfigurations as $code => $gridConfiguration) {
-            if (isset($gridConfiguration['extends'], $gridConfigurations[$gridConfiguration['extends']])) {
-                $gridConfiguration = $this->extend($gridConfiguration, $gridConfigurations[$gridConfiguration['extends']]);
+        $this->converter = $converter;
+        $this->gridConfigurations = $gridConfigurations;
+    }
+
+    private function convertGrids()
+    {
+        foreach ($this->gridConfigurations as $code => $gridConfiguration) {
+            if (isset($gridConfiguration['extends'], $this->gridConfigurations[$gridConfiguration['extends']])) {
+                $gridConfiguration = $this->extend($gridConfiguration, $this->gridConfigurations[$gridConfiguration['extends']]);
             }
 
-            $this->grids[$code] = $converter->convert($code, $gridConfiguration);
+            $this->grids[$code] = $this->converter->convert($code, $gridConfiguration);
         }
     }
 
+    public function reset()
```

Comment:
We can also add a `: void` typehint here :)

Reply:
Done :)